element_zstack/export/bossdb_interface.py

```diff
@@ -0,0 +1,209 @@
+            once.
+            retry_max (int, optional): Number of retries to upload a single
+            increment.
+            overwrite (bool, optional): Overwrite existing data.
+        """
+
+        self._url = url
+        self.url_bits = _parse_bossdb_uri(url)
+        self._volume_data = volume_data
+        self._voxel_size = tuple(float(i) for i in voxel_size)
+        self._voxel_units = voxel_units
+        self._data_description = data_description
+        self._upload_increment = upload_increment
+        self._retry_max = retry_max
+        self._overwrite = overwrite
+        self.description = "Uploaded via DataJoint"
+        self._resources = dict()
+        self._shape_zyx = tuple(int(i) for i in self._volume_data.shape)
+
+        try:
+            type(array(self._url))
+        except HTTPError:
+            self.url_exists = False
+        else:
+            self.url_exists = True
+        if not overwrite and self.url_exists:
+            logger.warning(
+                f"Dataset exists already exists at {self._url}\n"
+                + " To overwrite, set `overwrite` to True"
+            )
+            return
+
+        if not self.url_exists:
+            self.try_create_new()
+
+    def upload(self):
+        """Upload data to bossdb."""
+        z_max = self._shape_zyx[0]
+        boss_dataset = array(
+            self._url,
+            extents=self._shape_zyx,
+            dtype=str(self._volume_data.dtype),
+            voxel_size=self._voxel_size,
+            voxel_unit=self._voxel_units,
+        )
+        for i in tqdm(range(0, z_max, self._upload_increment)):
+            if i + self._upload_increment > self._shape_zyx[0]:
+                # We're at the end of the stack, so upload the remaining images.
+                images = self._volume_data[i : self._shape_zyx[0], :, :]
+                retry_count = 0
+                while True:
+                    try:
+                        boss_dataset[
+                            i : i + images.shape[0],
+                            0 : images.shape[1],
+                            0 : images.shape[2],
+                        ] = images
+                        break
+                    except Exception as e:
+                        print(f"Error uploading chunk {i}-{i + images.shape[0]}: {e}")
+                        retry_count += 1
+                        if retry_count > self._retry_max:
+                            raise e
+                        print("Retrying...")
+                        continue
+            else:
+                images = self._volume_data[i : i + self._upload_increment]
+                retry_count = 0
+                while True:
+                    try:
+                        boss_dataset[
+                            i : i + images.shape[0],
+                            0 : images.shape[1],
+                            0 : images.shape[2],
+                        ] = images
```

Comment:
I might be misunderstanding but should it be the following?

```suggestion
                        boss_dataset[
                            i : i + self._upload_increment,
                            0 : images.shape[1],
                            0 : images.shape[2],
                        ] = images
```

Reply:
My mistake.  Please disregard.